src/main/java/com/yelp/nrtsearch/server/luceneserver/logging/HitsLoggerCreator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.logging;
+
+import com.yelp.nrtsearch.server.config.LuceneServerConfiguration;
+import com.yelp.nrtsearch.server.grpc.LoggingHits;
+import com.yelp.nrtsearch.server.plugins.HitsLoggerPlugin;
+import com.yelp.nrtsearch.server.plugins.Plugin;
+import com.yelp.nrtsearch.server.utils.StructValueTransformer;
+
+/** Factory class that handles registration and creation of {@link HitsLogger}s. */
+public class HitsLoggerCreator {
+  private static HitsLoggerCreator instance;
+  private HitsLoggerProvider<?> hitsLoggerProvider;
+
+  /**
+   * Constructor.
+   *
+   * @param configuration server configuration
+   */
+  public HitsLoggerCreator(LuceneServerConfiguration configuration) {}
+
+  private void register(HitsLoggerProvider<?> hitsLoggerProvider) {
+    if (this.hitsLoggerProvider != null) {
+      throw new IllegalArgumentException("Hits logger already exists");
+    }
```

Comment:
This limitation would mean only one implementation of a logger can be active on a cluster at one time. That is pretty restrictive.

Is there a reason not use a mapping on name -> implementation, like we do for most the other extendable components? The query would then specify the implementation to use.

Reply:
Good question. I actually thought about the two solutions, but thinking about it again, I don't think being very restrictive here makes sense anymore. I thought that a single logger active on a cluster would be enough but being flexible and allowing multiple logger implementations is more beneficial, especially for clusters with multiple index. I will update the solution to support multiple loggers.